### PR TITLE
feat: Support PodDisruptionBudgets

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/435

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+You can configure the allowed Pod disruptions for HBase nodes as described in xref:concepts:operations/pod_disruptions.adoc[].
+
+Unless you configure something else or disable our PodDisruptionBudgets (PDBs), we write the following PDBs:
+
+== Master
+We only allow a single Master to be offline at any given time, regardless of the number of replicas or `roleGroups`.
+
+== RegionServer
+TODO
```
Comment (sbernauer) [inline docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc:11]: Marker for docs

File: rust/operator-binary/src/operations/pdb.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    Ok(())
+}
+
+fn max_unavailable_masters() -> u16 {
+    1
+}
+
+fn max_unavailable_region_servers() -> u16 {
+    // TODO: Let our expert Lars decide :)
+    1
```
Comment (sbernauer) [inline rust/operator-binary/src/operations/pdb.rs:66]: Marker: We need a decision here :) CC @lfrancke

File: rust/operator-binary/src/operations/pdb.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    #[snafu(display("Cannot apply PodDisruptionBudget [{name}]"))]
+    ApplyPdb {
+        source: stackable_operator::error::Error,
+        name: String,
+    },
+}
+
+pub async fn add_pdbs(
+    pdb: &PdbConfig,
+    zookeeper: &HbaseCluster,
```
Comment (sbernauer) [inline rust/operator-binary/src/operations/pdb.rs:26]: marker to fix

**Comment by sbernauer** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hbase-operator-it-custom/106/

File: docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc (hunk 1 above)
Comment (sbernauer) [reply docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc]: @lfrancke what do you think of the following sentence?

We only allow a single RegionServer to be offline at any given time, regardless of the number of replicas or `roleGroups`. HBase replicates the regions of your tables across `<replication-factor>` RegionServers, where you can configure the replication factor for every table individually. As we can't make any assumptions about the replication factors of your tables, we are conservative here to reduce downtime.

Comment (lfrancke) [reply docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc]: No, HBase doesn't replicate the regions (at least not by default, it has a feature called read replica but very few people use it).

Suggestion: To have as little data unavailable at any given time we only allow a single RegionServer to be unavailable.

Comment (sbernauer) [reply docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc]: Thanks! To stick to the theme we use in other operators, is the following also ok?

We only allow a single RegionServer to be offline at any given time, regardless of the number of replicas or `roleGroups`, to have as little data unavailable as possible at any given time.

Comment (lfrancke) [reply docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc]: Works for me, yes!

Comment (sbernauer) [reply docs/modules/hbase/pages/usage-guide/concepts/pod-disruptions.adoc]: Thanks!

File: docs/modules/hbase/pages/usage-guide/operations/graceful-shutdown.adoc [hunk 4]
```diff
@@ -0,0 +1,7 @@
+= Graceful shutdown
+
+Graceful shutdown of HBase nodes is either not supported by the product itself
+or we have not implemented it yet.
+
+The efforts of implementing graceful shutdown for all products that have this functionality is tracked in
```
Comment (adwk67) [inline docs/modules/hbase/pages/usage-guide/operations/graceful-shutdown.adoc:6]:
```suggestion
Outstanding implementation work for the graceful shutdowns of all products where this functionality is relevant is tracked in
```

File: docs/modules/hbase/pages/usage-guide/operations/index.adoc [hunk 5]
```diff
@@ -0,0 +1,5 @@
+= Operations
+
+This section of the documentation is intended for the operations teams that maintain a Stackable Data Platform installation.
+
+Please read on the xref:concepts:operations/index.adoc[Concepts page on Operations] with the necessary details to operate the platform in a production environment.
```
Comment (adwk67) [inline docs/modules/hbase/pages/usage-guide/operations/index.adoc:5]:
```suggestion
Please read the xref:concepts:operations/index.adoc[Concepts page on Operations] that contains the necessary details to operate the platform in a production environment.
```

File: docs/modules/hbase/pages/usage-guide/operations/pod-disruptions.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
+= Allowed Pod disruptions
+
+You can configure the allowed Pod disruptions for HBase nodes as described in xref:concepts:operations/pod_disruptions.adoc[].
```
Comment (adwk67) [inline docs/modules/hbase/pages/usage-guide/operations/pod-disruptions.adoc:3]:
```suggestion
You can configure the permitted Pod disruptions for HBase nodes as described in xref:concepts:operations/pod_disruptions.adoc[].
```

**Review by adwk67 — CHANGES_REQUESTED** [review]
lgtm % comments

**Review by adwk67 — APPROVED** [review]

**Pull request merged**